edx_user_state_client/tests.py

```diff
@@ -347,6 +379,46 @@ def test_delete_many_last_field(self):
         self.delete_many(user=0, blocks=[0, 1], fields=['a', 'b'])
         self.assertItemsEqual(self.get_many(user=0, blocks=[0, 1]), [])
 
+    def test_get_mod_date(self):
+        start_time = datetime.now()
+        self.set_many(user=0, block_to_state={0: {'a': 'b'}, 1: {'b': 'c'}})
+        end_time = datetime.now()
+
+        mod_dates = self.get_mod_date(user=0, block=0)
+
+        self.assertItemsEqual(mod_dates.keys(), ["a"])
+        self.assertGreater(mod_dates["a"], start_time)
+        self.assertLess(mod_dates["a"], end_time)
+
+    def test_get_mod_date_many(self):
+        start_time = datetime.now()
+        self.set_many(
+            user=0,
+            block_to_state={0: {'a': 'b'}, 1: {'b': 'c', 'c': 'd'}})
+        mid_time = datetime.now()
+        self.set_many(
+            user=0,
+            block_to_state={0: {'a': 'b2'}, 1: {'b': 'c2', 'c': 'd'}})
+        end_time = datetime.now()
+
+        mod_dates = [(key, field, mod)
+                     for key, field, mod
+                     in self.get_mod_date_many(
+                         user=0,
+                         blocks=[0, 1],
+                         fields=["b", "c"])]
+
+        self.assertItemsEqual(
+            [key for key, field, mod in mod_dates],
+            [self._block(1), self._block(1)])
+        self.assertItemsEqual(
+            [field for key, field, mod in mod_dates],
+            ["b", "c"])
+        self.assertGreater(mod_dates[0][2], mid_time)
+        self.assertLess(mod_dates[0][2], end_time)
```

Comment:
This is a bit hard to read as written, because of the double indexing. I think a namedtuple would help.

Also, the keys here assume a specific ordering of the results (blocks 0 and 1), which I don't think is specified by get_mod_date_many